### DISCUSSION TITLE
Welcome to Checked C Language Specification Discussions!

### DISCUSSION THREAD

**Discussion by sulekhark** [checkedc/checkedc#460] [Announcements]
## 👋 Welcome to the Checked C Language Specification Project!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  📢 **Announcements to the community from the maintainers** : We would like to reserve the "Announcements" category for official announcements related to the Checked C Language Specification.

**Comment by yakupjan01** [comment]
hello I am a beginner in c programming.
I want to know how can i learn will and use it will?thank you

**Reply by sulekhark** [reply]
Please read the Checked C language specification (checkedc-v0.9.pdf) available [here](https://github.com/Microsoft/checkedc-clang/releases), and start by writing small examples.

**Comment by IvanArkhipov1999** [comment]
Hello, I have some questions:

1. Why didn't you make all static arrays checked by default? I can't see any reason why it may be useful to use unchecked static array. From the point of view of compatibility with C, it seems, that checking the boundaries of static arrays should not cause any problems.
2. Sorry, if it is written in the document, but is it possible to make checked dynamic array? I haven't found any examples. Is it possible to write `int a checked[n][n]`?

**Comment by arunkumarbhattar** [comment]
I am trying to compile CheckedC library along with c++ file (which calls the library function). However, my clang (checked-C clang) compiler stops recognizing _Ptr and other "checked" jargon anymore. How do I fuse them together??.
NOTE:
1.) the same clang compiler compiles "only" the CheckedC libraries perfectly
2.) the same clang compiler compiles c++ file along with "non-checked" libraries perfectly.

Please help me!!

**Reply by sulekhark** [reply]
The CheckedC project is no longer under active development at Microsoft. Please post your questions here: 
https://github.com/secure-sw-dev/checkedc and https://github.com/secure-sw-dev/checkedc-llvm-project.

**Comment by Nausheen15** [comment]
Does Checked C optimize away unnecessary runtime checks using its own static analysis or does it offload it to the compiler optimization performed in later stages of the pipeline? 

Is there an easy way of finding out which of the Checked C runtime checks were proved and removed?

**Comment by SheldonHH** [comment]
How to see what's the length of converted N_Array_ptr and Array_ptr?

![image](https://github.com/microsoft/checkedc/assets/16319106/62e6da26-d048-4baa-b007-f94fc1283fd9)
